src/Lecture1.hs

```diff
@@ -98,7 +109,16 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString start end str =
+    if start > 0
+    then
+        let 
+
+            s = start -1
+            e = end - 1
+        in subString s e (tail str)
```

Comment:
`tail` is a partial function and throws an error on empty strings. So, this function will fail on empty `str` 😬 

I see that there's no such test case in unit tests for this exercise 🤔 Feel free to contribute such test directly to the course if you feel like doing this 🙂